editor/src/templates/editor.tsx

```diff
@@ -518,75 +503,44 @@ export class Editor {
-            }
+        Measure.taskTime(`Update Editor ${updateId}`, () => {
+          ReactDOM.flushSync(() => {
+            ReactDOM.unstable_batchedUpdates(() => {
+              Measure.taskTime(`Update Main Store ${updateId}`, () => {
+                this.utopiaStoreHook.setState(
+                  patchedStoreFromFullStore(this.storedState, 'editor-store'),
+                )
+              })
 
-            if (
-              shouldUpdateLowPriorityUI(this.storedState.strategyState, currentElementsToRender)
-            ) {
-              if (PerformanceMarks) {
-                performance.mark(`update low priority store ${updateId}`)
+              if (
+                shouldUpdateLowPriorityUI(
+                  this.storedState.strategyState,
+                  ElementsToRerenderGLOBAL.current,
+                )
+              ) {
+                Measure.taskTime(`Update Low Prio Store ${updateId}`, () => {
+                  this.lowPriorityStore.setState(
+                    patchedStoreFromFullStore(this.storedState, 'low-priority-store'),
+                  )
+                })
+              }
+              if (MeasureSelectors) {
+                logSelectorTimings('store update phase')
               }
-              this.lowPriorityStore.setState(
-                patchedStoreFromFullStore(this.storedState, 'low-priority-store'),
-              )
               if (PerformanceMarks) {
-                performance.measure(
-                  `Update Low Prio Store ${updateId}`,
-                  `update low priority store ${updateId}`,
-                )
+                performance.mark(`react wrap up ${updateId}`)
               }
-            }
-            const afterStoreUpdate = MeasureSelectors ? performance.now() : 0
-            if (MeasureSelectors) {
-              console.info(
-                'Dispatched actions:',
-                simpleStringifyActions(dispatchedActions),
-                'All stores',
-                afterStoreUpdate - beforeCanvasStore,
-                'Canvas store',
-                afterCanvasStore - beforeCanvasStore,
-                'main store',
-                afterMainStore - beforeMainStore,
-                'slow store',
-                afterStoreUpdate - afterMainStore,
-              )
-              logSelectorTimings('store update phase')
-            }
-            if (PerformanceMarks) {
-              performance.mark(`react wrap up ${updateId}`)
-            }
 
-            // reset selector timings right before the end of flushSync means we'll capture the re-render related selector data with a clean slate
-            resetSelectorTimings()
+              // reset selector timings right before the end of flushSync means we'll capture the re-render related selector data with a clean slate
+              resetSelectorTimings()
+            })
           })
         })
         if (PerformanceMarks) {
           performance.measure(
```

Comment:
I've kept this one as a manual measure because it measures is the amount of work React.flushSync does after the inner callback have returned.